### PR TITLE
initial commit of cloudfront integrations for s3 bucket static content

### PR DESCRIPTION
**Pull request by paulSambolin**

### CODE REVIEW

File: sweet-skills-infrastructure/web-hosting.template [hunk 1]
```diff
@@ -121,6 +126,46 @@
...
+        },
+
+        "getCertificates": {
+            "Type": "Custom::getCertificates",
+            "Properties": {
+                "ServiceToken": { "Fn::Join" : [ "", ["arn:aws:lambda:", { "Ref" : "AWS::Region" }, ":", { "Ref" : "AWS::AccountId" }, ":function:", { "Ref" : "GetCertificatesLambdaFunction" } ] ] }
+            }
+        },
+
+        "myDistribution" : {
```
Comment (ironmanwaring) [inline sweet-skills-infrastructure/web-hosting.template]: @paulSambolin 
can we call this something different?

File: sweet-skills-infrastructure/web-hosting.template [hunk 2]
```diff
@@ -121,6 +126,46 @@
                 "ServiceToken": { "Fn::Join" : [ "", ["arn:aws:lambda:", { "Ref" : "AWS::Region" }, ":", { "Ref" : "AWS::AccountId" }, ":function:", { "Ref" : "S3LambdaFunction" } ] ] },
                 "BucketName" : { "Ref" : "FullDomainName"}
             }
+        },
```
Comment (ironmanwaring) [inline sweet-skills-infrastructure/web-hosting.template]: @paulSambolin should the certificate and cloudfront resources be in the web hosting template or web routing?

File: sweet-skills-infrastructure/_full-stack.template [hunk 3]
```diff
@@ -288,6 +294,13 @@
             }
         },
 
+        "ApiPublicIp": {
```
Comment (ironmanwaring) [inline sweet-skills-infrastructure/_full-stack.template]: @paulSambolin why did this need to be taken out of the app-server template?

File: sweet-skills-infrastructure/web-routing.template [hunk 4]
```diff
@@ -95,8 +104,8 @@
                 "Name": { "Fn::Join" : ["", [ "www.", { "Ref": "FullDomainName" } ]] },
                 "Type": "A",
                 "AliasTarget": {
-                    "DNSName": { "Fn::FindInMap" : [ "Regions", { "Ref": "AWS::Region" }, "s3DnsName"] },
-                    "HostedZoneId": { "Fn::FindInMap" : [ "Regions", { "Ref": "AWS::Region" }, "hostedZoneId"] }
+                    "DNSName": { "Ref": "CloudFrontDomainName" },
+                    "HostedZoneId": { "Ref" : "CloudFrontHostedZone" }
                 },
                 "HostedZoneId": { "Ref": "HostedZoneId" }
             }
```
Comment (ironmanwaring) [inline sweet-skills-infrastructure/web-routing.template]: @paulSambolin is there a reason we're leaving the route 53 record sets as is?  Are they still being used the same way or should they be modified/removed?

**Comment by paulSambolin** [comment]
@ironmanwaring Made changes and tested successfully

**Comment by ironmanwaring** [comment]
lgtm

**Pull request merged**